File: golden-paths/elixir/phoenix/test/deliberate_test.exs
defmodule GoldenPhoenix.DeliberateTest do
  # Deliberately-failing test — proves the elixir lane PROPAGATES failure. It carries @tag :selfcheck;
  # test_helper excludes that tag from a normal run and the lane self-check runs ONLY it via
  # `mix test --only selfcheck`. Tag-based (not a name filter) → fail-closed on a rename.
  use ExUnit.Case

  @tag :selfcheck
  test "deliberate failure" do
    flunk("selfcheck: the golden-path elixir lane must surface this failure (exit non-zero)")
  end
end

File: golden-paths/clojure/ring/test/golden/deliberate_test.clj
(ns golden.deliberate-test
  ;; Deliberately-failing test — proves the clojure lane PROPAGATES failure. It carries ^:selfcheck; the
  ;; :default test-selector (in project.clj) excludes it from a normal run and `lein test :selfcheck`
  ;; runs ONLY it. Metadata-based (not a name filter) → fail-closed on a rename.
  (:require [clojure.test :refer [deftest is]]))

(deftest ^:selfcheck deliberate-failure
  (is false "selfcheck: the golden-path clojure lane must surface this failure (exit non-zero)"))
